### DISCUSSION TITLE
Which internal mechanisms should be still DLL-exported in vNext?

### DISCUSSION THREAD

**Discussion by frederick-vs-ja** [microsoft/STL#3871] [General] [answered]
There're several issues saying that some internal mechanisms shouldn't be exported in vNext. Conversely, it seems that only a few internal mechanism should be exported.

**Comment by StephanTLavavej** [comment, marked as answer]
We should DLL-export (with flat C functions only) mechanisms that are still used by the headers, while we shouldn't DLL-export things that aren't used by the headers. (They should either remain purely internal to the STL's DLL and static LIB - we currently have a number of functions that are used only for cross-TU "communication" within the STL's DLL and static LIB and are uselessly DLL-exported despite not being mentioned by the headers at all - or they should be identified as completely dead code and outright eliminated.)
